R/submit-model.R

```diff
@@ -62,75 +73,6 @@ submit_model.bbi_nmbayes_model <- function(
              .dry_run = .dry_run)
 }
 
-#' Run Bayes chains
-#'
-#' Run multiple chains of a Bayes model after initial estimates have been
-#' generated
-#'
-#' @param .mod A `bbi_nmbayes_model` object.
-#' @param ... Arguments passed to [bbr::submit_model()].
-#' @noRd
-run_chains <- function(.mod, ...) {
-  checkmate::assert_class(.mod, NMBAYES_MOD_CLASS)
-
-  ctl <- readr::read_lines(get_model_path(.mod))
-
-  row_bayes <- stringr::str_detect(ctl, "METHOD=BAYES|METHOD=NUTS")
-  est_bayes <- ctl[row_bayes]
-  est_bayes <- stringr::str_replace(est_bayes, "^;", "")
-  ctl[row_bayes] <- est_bayes
-
-  row_table <- stringr::str_detect(ctl, ";\\s*\\$TABLE")
```

Comment:
I'm seeing unexpected (to me) behavior with the `$TABLE` statement when using `copy_model_as_nmbayes()`

(I'm trying, and failing, to find the code where this is happening, so instead I'm commenting on this code where it appears it _used to_ happen.)

Basically, when I do `model_diff()` I see that the `$TABLE` statement from the parent has been removed. Is that intentional? I don't see that documented anywhere (and I'm not clear on the reason for it, though it may be totally reasonable). Am I missing something?

<img width="1084" alt="Screenshot 2023-06-15 at 4 20 12 PM" src="https://github.com/metrumresearchgroup/bbr.bayes/assets/7806083/2a7bf26b-a4df-4e9d-91a7-864bb0900c32">

Even weirder (to me), I'm _not_ seeing this behavior when I copy from a "regular" NONMEM model:

<img width="1079" alt="Screenshot 2023-06-15 at 4 47 20 PM" src="https://github.com/metrumresearchgroup/bbr.bayes/assets/7806083/98e10128-a92a-44f4-91a2-5346aa133feb">

Reply:
> when I do model_diff() I see that the $TABLE statement from the parent has been removed. Is that intentional?

No, it's not.  Thanks for the snippets.  I'll look into it.

Reply:
> I see that the `$TABLE` statement from the parent has been removed

Looking at that more closely, that's a comment after an `$EST ...` record.   `nmrec` treats any comments following a record as part of the preceding record, so the `; $TABLE` is considered part of the `$EST` record.  While I think that's reasonable behavior from nmrec's standpoint (comments are just blobs of text that have to be grouped with either the below or above record, and the above is simplest from an implementation standpoint), that's confusing in this case.  

I think it makes sense to preserve all trailing comments here, so I'll rework `copy_model_as_nmbayes`s approach.

Reply:
Ah ok, that makes sense. I hadn't even noticed that `$TABLE` that got removed was commented out. In that case, I think it's reasonable either way, but probably preserving them is the better move. Thanks for making the update.